stdlib/Distributed/src/remotecall.jl

```diff
@@ -643,3 +643,17 @@ function getindex(r::RemoteChannel, args...)
     end
     return remotecall_fetch(getindex, r.where, r, args...)
 end
+
+function Base.iterate(c::RemoteChannel, state=nothing)
+    try
+        return (take!(c), nothing)
+    catch e
+        if isa(e, InvalidStateException) && e.state == :closed
```

Comment:
`isa(e, InvalidStateException)` is not enough.

We'd need to handle the case when the channel is a reference

```
julia> ch = RemoteChannel(workers()[end])
RemoteChannel{Channel{Any}}(2, 1, 8)

julia> put!(ch, 1)
RemoteChannel{Channel{Any}}(2, 1, 8)

julia> close(ch)

julia> take!(ch)
1

julia> err = try
           take!(ch)
       catch err
           err
       end
RemoteException(2, CapturedException(InvalidStateException("Channel is closed.", :closed), Any[(check_channel_state at channels.jl:170 [inlined], 1), (take_buffered at channels.jl:388, 1), (take! at channels.jl:383, 1), (take! at remotecall.jl:599,
 1), (take_ref at remotecall.jl:610, 1), (#106 at process_messages.jl:278, 1), (run_work_thunk at process_messages.jl:63, 1)
, (macro expansion at process_messages.jl:278 [inlined], 1), (#105 at task.jl:411, 1)]))
```